internal/extproc/watcher.go

```diff
@@ -66,9 +68,56 @@ func (cw *configWatcher) loadConfig() error {
 	}
 	cw.lastMod = stat.ModTime()
 	cw.l.Info("loading a new config", slog.String("path", cw.path))
+
+	// Re-hydrate the current config file for later diffing.
+	previous := cw.current
+	current, err := cw.getConfig()
+	if err != nil {
+		return fmt.Errorf("failed to read the config file: %w", err)
+	}
+
+	// Print the diff between the old and new config.
+	if cw.l.Enabled(ctx, slog.LevelDebug) {
+		cw.diff(previous, current)
+	}
+
 	cfg, err := filterconfig.UnmarshalConfigYaml(cw.path)
 	if err != nil {
 		return err
 	}
 	return cw.rcv.LoadConfig(cfg)
 }
+
+func (cw *configWatcher) getConfig() (string, error) {
+	currentByte, err := os.ReadFile(cw.path)
```

Comment:
this overlaps with `filterconfig.UnmarshalConfigYaml(cw.path)` above. can you do the deduplication?